packages/klighd-core/src/titles/title-storage.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * KIELER - Kiel Integrated Environment for Layout Eclipse RichClient
+ *
+ * http://rtsys.informatik.uni-kiel.de/kieler
+ *
+ * Copyright 2023 by
+ * + Kiel University
+ *   + Department of Computer Science
+ *     + Real-Time and Embedded Systems Group
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+import { VNode } from 'snabbdom';
+import { Transformation } from '../views-common';
+
+/**
+ * Storage of data related to title renderings with the 'klighd.isNodeTitle' property.
+ * Requires the user to call {@link decendToChild()} when starting the rendering of a child graph element and {@link ascendToParent()}
```

Comment:
is there any way we can enforce this contract? As I understand it now this is solely the  responsibilty of the programmer.

Reply:
enforced it by calling these methods at a more central location during the model rendering.